### PR TITLE
6922 - Freeze entity in COLIN

### PR DESCRIPTION
**Pull request by andrepestana-aot**
*Issue #:* /bcgov/entity#6922

*Description of changes:*

- Updated the corp_frozen_typ_cd for Alterations from legal-api

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

File: colin-api/src/colin_api/models/filing.py [hunk 1]
```diff
@@ -863,6 +863,10 @@ def add_filing(cls, con, filing: Filing) -> int:
                 Business.update_corporation(
                     cursor=cursor, corp_num=corp_num, date=agm_date, annual_report=is_annual_report)
 
+                # Freeze entity for Alteration
+                if filing.filing_type == 'alteration':
+                    Business.update_corp_frozen_type(cursor, corp_num, Business.CorpFrozenTypes.COMPANY.value)
```
Comment (andrepestana-aot) [inline colin-api/src/colin_api/models/filing.py:868]: As discussed with @Sienna-Blumstengel and @thorwolpert , if it is an affiliation we don't need to check the corp type and freeze the corporation.

**Review by kialj876 — APPROVED** [review]
Looks good to me

File: colin-api/src/colin_api/models/business.py [hunk 2]
```diff
@@ -46,6 +46,12 @@ class TypeCodes(Enum):
         BC_COMP = 'BC'
         ULC_COMP = 'ULC'
 
+    class CorpFrozenTypes(Enum):
```
Comment (thorwolpert) [inline colin-api/src/colin_api/models/business.py:49]: Could you expand on the description of what the codes are.

**Review by thorwolpert — COMMENTED** [review]
Please add a description of the frozen codes, as their use is not easily understood from the names

File: colin-api/src/colin_api/models/business.py (hunk 2 above)
Comment (andrepestana-aot) [reply colin-api/src/colin_api/models/business.py:49]: Yes, good point! Thanks!

Comment (andrepestana-aot) [reply colin-api/src/colin_api/models/business.py:49]: I've changed it.

Comment (thorwolpert) [reply colin-api/src/colin_api/models/business.py:49]: I don't see anything, maybe a missing push?
when would I choose, or use:
```
        COMPANY_FROZEN = 'C'
        SELF_SERVED_FROZEN = 'S'
```

**Review by thorwolpert — COMMENTED** [review]
i didn't see a comment on the different ```CorpFrozenTypes```

File: colin-api/src/colin_api/models/filing.py [hunk 3]
```diff
@@ -863,6 +863,10 @@ def add_filing(cls, con, filing: Filing) -> int:
                 Business.update_corporation(
                     cursor=cursor, corp_num=corp_num, date=agm_date, annual_report=is_annual_report)
 
+                # Freeze entity for Alteration
+                if filing.filing_type == 'alteration':
+                    Business.update_corp_frozen_type(cursor, corp_num, Business.CorpFrozenTypes.COMPANY_FROZEN.value)
```
Comment (andrepestana-aot) [inline colin-api/src/colin_api/models/filing.py:868]: Here's the change to COMPANY_FROZEN

File: colin-api/src/colin_api/models/business.py (hunk 2 above)
Comment (andrepestana-aot) [reply colin-api/src/colin_api/models/business.py:49]: Please see https://github.com/bcgov/lear/pull/1138/files#r611738801

Comment (thorwolpert) [reply colin-api/src/colin_api/models/business.py:49]: I'm looking for something like:

```
 """Render an Enum of the Corporation Frozen Type Codes.

     The following frozen states are used for:
     COMPANY_FROZEN is when the Registrar has frozen a company
     SELF_SERVED_FROZEN is when the company has ????
 """
 ```

**Review by thorwolpert — COMMENTED** [review]
i put in an example of what I meant by "Could you expand on the description of what the codes are."

File: colin-api/src/colin_api/models/business.py (hunk 2 above)
Comment (andrepestana-aot) [reply colin-api/src/colin_api/models/business.py:49]: SELF_SERVED_FROZEN  is a legacy code and is deprecated. I updated the comment.

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**